Azure transcription service improvements

wsmirnow opened the pull request:
This is a successor of https://github.com/opencast/opencast/pull/4118 as the original PR has merge conflicts. It was easier to rewrite the patch with all review comments resolved as to merge the original one.

This patch address the misuse of Java temp dir in Microsoft Azure transcription service. The Opencast workspace is designed for this kind of work and we should use it.

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

wsmirnow committed:
Transcription service Azure improvements

The Microsoft Azure transcription service should not make use of Java temp directory but put all contents into workspace instead. Also some logging messages are improved.

wsmirnow committed:
Microsoft Azure transcription service documentation improvements

Arnei reviewed (approved):
Code looks good to me and runs.

wsmirnow committed:
Captions filename fixed

wsmirnow committed:
Transcription service azure possible racecondition fixed

Two or more transcription jobs can override detected language result for each other as the result is stored in a global variable. Storing the detected language result in a file solves the issue.
